pkg/apis/openliberty/v1beta1/openlibertyapplication_types.go

```diff
@@ -468,6 +533,167 @@ func (m *OpenLibertyApplicationMonitoring) GetEndpoints() []prometheusv1.Endpoin
+	return o.UserNameAttribute
+}
+
+// GetDisplayName returns displayName
+func (o *OidcClient) GetDisplayName() string {
+	return o.DisplayName
+}
+
+// GetUserInfoEndpointEnabled returns userInfoEndpointEnabled
+func (o *OidcClient) GetUserInfoEndpointEnabled() *bool {
+	return o.UserInfoEndpointEnabled
+}
+
+// GetRealmNameAttribute returns realmNameAttribute
+func (o *OidcClient) GetRealmNameAttribute() string {
+	return o.RealmNameAttribute
+}
+
+// GetScope returns scope
+func (o *OidcClient) GetScope() string {
+	return o.Scope
+}
+
+// GetTokenEndpointAuthMethod returns tokenEndpointAuthMethod
+func (o *OidcClient) GetTokenEndpointAuthMethod() string {
+	return o.TokenEndpointAuthMethod
+}
+
+// GetHostNameVerificationEnabled returns hostNameVerificationEnabled
+func (o *OidcClient) GetHostNameVerificationEnabled() *bool {
+	return o.HostNameVerificationEnabled
+}
+
+// GetID returns ID of oauth2Login (specified in Liberty's server.xml). Default value is "oauth2"
+func (o *OAuth2Client) GetID() string {
+	if o.ID == "" {
+		return "oauth2"
+	}
+	return o.ID
+}
+
+// GetTokenEndpoint returns tokenEndpoint
+func (o *OAuth2Client) GetTokenEndpoint() string {
+	return o.TokenEndpoint
+}
+
+// GetAuthorizationEndpoint returns authorizationEndpoint
+func (o *OAuth2Client) GetAuthorizationEndpoint() string {
+	return o.AuthorizationEndpoint
+}
+
+// GetGroupNameAttribute returns groupNameAttribute
+func (o *OAuth2Client) GetGroupNameAttribute() string {
+	return o.GroupNameAttribute
+}
+
+// GetUserNameAttribute returns userNameAttribute
+func (o *OAuth2Client) GetUserNameAttribute() string {
+	return o.UserNameAttribute
+}
+
+// GetDisplayName returns displayName
+func (o *OAuth2Client) GetDisplayName() string {
+	return o.DisplayName
+}
+
+// GetRealmNameAttribute returns realmNameAttribute
+func (o *OAuth2Client) GetRealmNameAttribute() string {
+	return o.RealmNameAttribute
+}
+
+// GetRealmName returns realmName
+func (o *OAuth2Client) GetRealmName() string {
+	return o.RealmName
+}
+
+// GetScope returns scope
+func (o *OAuth2Client) GetScope() string {
+	return o.Scope
+}
+
+// GetTokenEndpointAuthMethod returns tokenEndpointAuthMethod
+func (o *OAuth2Client) GetTokenEndpointAuthMethod() string {
+	return o.TokenEndpointAuthMethod
+}
+
+// GetAccessTokenHeaderName returns accessTokenHeaderName
+func (o *OAuth2Client) GetAccessTokenHeaderName() string {
+	return o.AccessTokenHeaderName
+}
+
+// GetAccessTokenRequired returns accessTokenRequired
+func (o *OAuth2Client) GetAccessTokenRequired() *bool {
+	return o.AccessTokenRequired
+}
+
+// GetAccessTokenSupported returns accessTokenSupported
+func (o *OAuth2Client) GetAccessTokenSupported() *bool {
+	return o.AccessTokenSupported
+}
+
+// GetUserApiType returns userApiType
+func (o *OAuth2Client) GetUserApiType() string {
+	return o.UserApiType
+}
+
+// GetUserApi returns userApi
+func (o *OAuth2Client) GetUserApi() string {
+	return o.UserApi
+}
+
```

Comment:
I know its trying to be consistent with BaseComponent interface impl. But having getter on simple structs (not interface) is redundant and is anti-pattern in Go

Reply:
Good point. Addressed.